wireless_modules/middle_module/boot.py

```diff
@@ -0,0 +1,54 @@
+# Author: Kunj
+# Last updated: 1/08/2020
+
+# Import relevant libraries
+import esp
+import network
+import config
+import time
+
+
+def do_connect(essid, password):
+    """
+    Tries to connect to the WIFI 5 times at max
+    :param essid: The WIFI name to connect to, ensure it's not in double quotes but single ones
+    :param password: The WIFI password to connect, ensure it's not in double quotes but single quotes
+    :return: None
+    """
+
+    max_wifi_attempts = 5
+
+    # Create station interface
+    station_interface = network.WLAN(network.STA_IF)
+    station_interface.active(True)
+
+    if station_interface.isconnected():
+        print('network config:', station_interface.ifconfig())
+
+        # Turn on the built in LED when booted and connected to WiFi (blue light)
+        config.builtin_LED.on()
+        time.sleep(5)
```

Comment:
If we do this checking, I guess we don't need boot.py to infinitely try and connect to the WIFI, it can be done through other files that check that the interface is connected and if not calls on the do_connect() method

Reply:
On that note, I can create a class in here with two methods, one to connect (or attempt to) and another that can be called to verify the connection is still established, what do you think?

Reply:
Yeah that sounds like a good idea! Maybe put the class in a new file - importing the boot module from our main code feels a bit wrong to me